src/rgw/rgw_sal_motr.cc

```diff
@@ -3493,7 +3493,30 @@ int MotrStore::list_users(const DoutPrefixProvider* dpp, const std::string& meta
                         std::string& marker, int max_entries, void *&handle,
                         bool* truncated, std::list<std::string>& users)
 {
-    return 0;
+  int rc;
+  bool max_entires_specified = true;
+  if(max_entries <= 0) {
+    max_entries = 1000; 
+    max_entires_specified = false;
+  }
+  vector<string> keys(max_entries + 1);
+  vector<bufferlist> vals(max_entries + 1);
+
+  keys[0] = marker;
+  rc = next_query_by_name(RGW_MOTR_USERS_IDX_NAME, keys, vals);
+  if (rc < 0) {
+    ldpp_dout(dpp, 0) << "ERROR: NEXT query failed. " << rc << dendl;
+    return rc;
+  }
+  if(!(keys.back()).empty() && max_entires_specified){
+    *truncated = true;
+  }
+  for(int i = 0; i < int(keys.size()) - 1; i++){
+    if(!keys[i].empty()){
+      users.push_back(keys[i]);
+    }
+  }
```

Comment:
If we have 2 entries & max-entries = 1000, then this loop will iterate 1000 times.
so either we need to break it in else condition 
OR 
I will suggest that we can use the `rc` which will be equal to num of keys fetched.
so if the result is truncated we will have to do rc-- to copy correct num of keys.
```c++
  if (keys.back()) {
    *truncated = true;
    --rc;
  }
  
  // copy function
  std::copy(keys.begin(), keys.begin() + rc, std::back_inserter(users));
  
  //  OR a loop
  for (int i = 0; i < rc; i++)
    users.push_back(keys[i])
```

Reply:
Good point. I have updated the loop body, it will break out now with first empty value.